### ISSUE TITLE
`ObservableSet.intersection` triggers "Maximum call stack size exceeded" stack overflow

### ISSUE THREAD

**Issue by travismfrank** [mobxjs/mobx#3919]
**Intended outcome:**

Calling the newly implemented `ObservableSet.intersection` with another `ObservableSet` as the argument causes a stack overflow error.  This is because an `ObservableSet` argument passes `isES6Set`, thus calling `intersection` on _that_ set with the first set as the argument.  Repeat ad infinitum until V8 decides it's had enough.

`observableSet.intersection(otherObservableSet)` should just work.

**Actual outcome:**

"Maximum call stack size exceeded" in Chrome.  Stack overflow.

**How to reproduce the issue:**

```typescript
const foo = observable.set(['a', 'b', 'c']);
const bar = observable.set(['c', 'd', 'e']);

const shared = foo.intersection(bar);
```

[Here's](https://codesandbox.io/p/sandbox/amazing-pare-mr29k6?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm0pun1gr00062e6atpvk2y66%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm0pun1gr00022e6aod7z0auw%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm0pun1gr00032e6ayeno9tak%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm0pun1gr00052e6a3qmnkc10%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm0pun1gr00022e6aod7z0auw%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm0pun1gr00012e6anp6vtp9k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A3%252C%2522startColumn%2522%253A28%252C%2522endLineNumber%2522%253A3%252C%2522endColumn%2522%253A28%257D%255D%257D%255D%252C%2522id%2522%253A%2522cm0pun1gr00022e6aod7z0auw%2522%252C%2522activeTabId%2522%253A%2522cm0pun1gr00012e6anp6vtp9k%2522%257D%252C%2522cm0pun1gr00052e6a3qmnkc10%2522%253A%257B%2522id%2522%253A%2522cm0pun1gr00052e6a3qmnkc10%2522%252C%2522activeTabId%2522%253A%2522cm0puvey3000j2e69otvyh3pg%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm0pun1gr00042e6ae6b1xbl2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522cm0puvey3000j2e69otvyh3pg%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522cm0pun1gr00032e6ayeno9tak%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cm0pun1gr00032e6ayeno9tak%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Afalse%252C%2522sidebarPanelSize%2522%253A15%257D) an interactive version, just click the button.

**Versions**

`mobx@6.13.1`

**Comment by urugator** [comment]
Possibly broken by https://github.com/mobxjs/mobx/pull/3893/files . It introduced new `isPlainES6Map`, but not `isPlainES6Set`.

**Comment by urugator** [comment]
should be fixed in mobx@6.13.2
